Error in windows 10 inkstitch v2.0

After the installation of the new version of inkstitch with the antivirus deactivated i got this error and i tried also to install again the inkskape the latest version of it and i have the same error...what could go wrong ...


Ink/Stitch experienced an unexpected error. 

If you'd like to help, please file an issue at https://github.com/inkstitch/inkstitch/issues and include the entire error description below:

Ink/Stitch Version: v2.0.0 (windows)


Traceback (most recent call last):
  File "inkstitch.py", line 52, in <module>
  File "inkex\base.py", line 141, in run
  File "lib\extensions\lettering.py", line 420, in effect
  File "lib\extensions\lettering.py", line 48, in __init__
  File "lib\extensions\lettering.py", line 151, in set_font_list
wx._core.wxAssertionError: C++ assertion "strcmp(setlocale(0, 0), "C") == 0" failed at ..\..\src\common\intl.cpp(1694) in wxLocale::GetInfo(): You probably called setlocale() directly instead of using wxLocale and now there is a mismatch between C/C++ and Windows locale.
Things are going to break, please only change locale by creating wxLocale objects to avoid this!

For me on Windows bisede this, one on lettering i also  got this on i try to do parameters...

Traceback (most recent call last):
  File "inkstitch.py", line 52, in <module>
  File "inkex\base.py", line 141, in run
  File "lib\extensions\params.py", line 626, in effect
  File "lib\extensions\params.py", line 340, in __init__
  File "lib\extensions\params.py", line 606, in create_tabs
  File "lib\extensions\params.py", line 69, in __init__
wx._core.wxAssertionError: C++ assertion "strcmp(setlocale(0, 0), "C") == 0" failed at ..\..\src\common\intl.cpp(1694) in wxLocale::GetInfo(): You probably called setlocale() directly instead of using wxLocale and now there is a mismatch between C/C++ and Windows locale.
Things are going to break, please only change locale by creating wxLocale objects to avoid this!

Hello, since updating to InkStitch 2.0.0 (windows) and trying to open the "Params" window I've been getting the same error:

```
Ink/Stitch Version: v2.0.0 (windows)

Traceback (most recent call last):
  File "inkstitch.py", line 52, in <module>
  File "inkex\base.py", line 141, in run
  File "lib\extensions\params.py", line 626, in effect
  File "lib\extensions\params.py", line 340, in __init__
  File "lib\extensions\params.py", line 606, in create_tabs
  File "lib\extensions\params.py", line 69, in __init__
wx._core.wxAssertionError: C++ assertion "strcmp(setlocale(0, 0), "C") == 0" failed at ..\..\src\common\intl.cpp(1694) in wxLocale::GetInfo(): You probably called setlocale() directly instead of using wxLocale and now there is a mismatch between C/C++ and Windows locale.
Things are going to break, please only change locale by creating wxLocale objects to avoid this!
```

Changing the language of my windows system fixed this issue. I reverted the language change to confirm it.

Previous language settings (caused the error):
- Windows: English (Austria)
- InkStitch: English
- InkScape: English

Current language settings (fixed the error):
- Windows: English (USA)
- InkStitch: English
- InkScape: English

Thanks for this thread, I would have never been able to fix this on my own in a million years. Happy stitching! :)

--- a/lib/extensions/lettering.py
+++ b/lib/extensions/lettering.py
@@ -30,6 +30,11 @@ class LetteringFrame(wx.Frame):
     DEFAULT_FONT = "small_font"
 
     def __init__(self, *args, **kwargs):
+        # This is necessary because of https://github.com/inkstitch/inkstitch/issues/1186
+        if sys.platform.startswith('win'):
+            import locale
+            locale.setlocale(locale.LC_ALL, "C")
+
         # begin wxGlade: MyFrame.__init__
         self.group = kwargs.pop('group')
         self.cancel_hook = kwargs.pop('on_cancel', None)

--- a/lib/extensions/params.py
+++ b/lib/extensions/params.py
@@ -330,6 +330,11 @@ class ParamsTab(ScrolledPanel):
 
 class SettingsFrame(wx.Frame):
     def __init__(self, *args, **kwargs):
+        # This is necessary because of https://github.com/inkstitch/inkstitch/issues/1186
+        if sys.platform.startswith('win'):
+            import locale
+            locale.setlocale(locale.LC_ALL, "C")
+
         # begin wxGlade: MyFrame.__init__
         self.tabs_factory = kwargs.pop('tabs_factory', [])
         self.cancel_hook = kwargs.pop('on_cancel', None)

